r/tests/testthat/helper-arrow.R

```diff
@@ -18,6 +18,8 @@
 # Wrap testthat::test_that with a check for the C++ library
 options(..skip.tests = !arrow:::arrow_available())
 
+set.seed(1)
```

Comment:
You could argue that some randomness is good (i.e. to remove this line) because it helped us find this issue (even though it was a heisenfailure)

Reply:
You could argue that, but I'd rather have that randomness somewhere other than our unit test suite.